qiskit_ibm/ibm_provider.py

```diff
@@ -320,21 +282,22 @@ def _initialize_providers(
                 default_provider=credentials.default_provider,
                 **hub_info
             )
-            provider_credentials.preferences = \
-                preferences.get(provider_credentials.unique_id(), {})
-            # _credentials class variable is read in __init__ to set provider credentials
-            cls._credentials = provider_credentials
-            # Build the provider.
+            hgp_credentials.preferences = \
+                preferences.get(hgp_credentials.unique_id(), {})
+            # Build the backend group.
             try:
-                provider = IBMProvider(token=credentials.token, **hub_info)
-                cls._providers[provider.credentials.unique_id()] = provider
-                # Clear _credentials class variable so __init__ is not processed for the first
-                # call to __new__ (since all IBMProvider instances are initialized by this method)
-                cls._credentials = None
+                backend_group = BackendGroup(credentials=hgp_credentials, provider=self)
+                self._backend_groups[backend_group.credentials.unique_id()] = backend_group
+                self._backend_group = backend_group
             except Exception:  # pylint: disable=broad-except
-                # Catch-all for errors instantiating the provider.
-                logger.warning('Unable to instantiate provider for %s: %s',
+                # Catch-all for errors instantiating the backend group.
+                logger.warning('Unable to instantiate hub/group/project for %s: %s',
                                hub_info, traceback.format_exc())
+        if credentials.default_provider:
+            hub, group, project = credentials.default_provider.to_tuple()
+            self._backend_group = self._get_backend_group(hub=hub, group=group, project=project)
+        elif len(self._backend_groups) == 1:
+            self._backend_group = self._get_backend_group()
```

Comment:
Since `_backend_groups` is already an ordered dictionary, do you really need to have a "default" `_backend_group`? Can't you just put the `default_provider` group as the first, and open access group the last. Then when you need something (e.g. service and/or backend), just loop through all of them in order until you find the right one.